tests/crypto/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+cmake_minimum_required(VERSION 3.8.2)
+include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
+
+project(NONE)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+FILE(GLOB app_sources src/*.c)
+
+if(NOT OVERLAY_CONFIG)
+  message(FATAL_ERROR "You must pass -DOVERLAY_CONFIG={overlay-backend-sw.conf|overlay-backend-hw.conf|overlay-backend-hw-sw.conf}")
+endif()
```

Comment:
Why doing this ?
It prevents users from using a well know pattern when controlling the conf files, such as:
`cmake -DBOARD=... -DCONF_FILE="prj.conf;overlay-backend-hw.conf" ..`
or with `west`:
`west -b<board> -- -DCONF_FILE="prj.conf;overlay-backend-hw.conf"`

The use of overlay file can be described in doc, as it is done in this example: 
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/samples/net/mqtt_publisher/README.html#tls-offloading

Reply:
The link you described uses the same pattern as this PR I think?

> It is also possible to connect to the MQTT broker through a SOCKS5 proxy. To enable it, use -DOVERLAY_CONFIG=overlay-socks5.conf when running west build or cmake.

I was not aware you could separate the config files by semi-colon.
I can change to use that pattern instead if it is clearer.
The fatal error is there since one of the overlays must be used, and a clear message is displayed when not.

Reply:
No, we should not force the user to use a specific overlay file.
He could write a custom conf files and use that instead if he has special wishes.
He can use: `-DCONF_FILE="mycomplete.conf"`, or `-DCONF_FILE="prj.conf;overlay-backend-sw.conf"`, or as you suggest: ``-DOVERLAY_FILE="overlay-backend-sw.conf"``.
So don't test the way you do.

If there a special values that must be set for the test to work, then test for that, do not test that a special file is provide.

Reply:
Changed in d3ca1e4